client/src/lib/dependency-injection/withInjector.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+function withInjector(Component, dependencies, mapDependenciesToProps) {
+  const injectorContext = {
+    injector: React.PropTypes.shape({
+      get: React.PropTypes.func,
+    }),
+  };
+
+  class Injectable extends React.Component {
+    render() {
+      let props = {};
+      if (dependencies) {
+        if (!Array.isArray(dependencies)) {
+          throw new Error(`
+            withInjector() passed an argument for dependencies that is ${typeof dependencies}. 
+            Must be an array of named dependencies.
+          `);
+          return;
+        }
+        const resolved = dependencies.map(this.context.injector.get);
```

Comment:
Yes. The idea is that almost everything should be wrapped in `withInjector()`, particularly containers, as they by definition have dependencies. Components that shouldn't would be simple presentational components that render only HTML. e.g. `const Title = (props) => <h2>{props.children}</h2>`;

There's nothing to my knowledge that interferes with an injector-consuming component being customised.